### PR TITLE
Offer sidebar

### PR DESCRIPTION
**Pull request by gustafguner**
Lots of stuff in this PR!

- Above the fold
- Sidebar
- Discount code input
- Perils insurance values

I have also created some nice generic components and improved the mobile experience.

There are some stuff regarding the sidebar that I've left out/mocked intentionally. Such as "view details" of the insurance. 

Screenshot 2019-11-29 at 11 38 07

Screenshot 2019-11-29 at 11 39 42

Screenshot 2019-11-29 at 11 38 38

Screenshot 2019-11-29 at 11 45 40

### CODE REVIEW

File: src/new-components/Tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import styled from '@emotion/styled'
+import { colorsV2 } from '@hedviginsurance/brand'
+import { Questionmark } from 'components/icons/Questionmark'
+import { motion } from 'framer-motion'
+import * as React from 'react'
+
+interface Props {
+  body: string
+  size?: 'sm' | 'lg'
```
Comment (hedvigoscar) [inline src/new-components/Tooltip.tsx]: Let's define this thing as a type 😉 

Something like

```typescript
type Size = 'sm' | 'lg'

interface Sized {
  size?: Size
}
```

File: src/new-components/buttons.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  size?: 'sm' | 'lg'
+  disabled?: boolean
+}>`
+  font-size: ${(props) => (props.size === 'lg' ? `1rem` : `0.875rem`)};
+  line-height: ${(props) => (props.size === 'lg' ? `1rem` : `1.25rem`)};
+  padding: ${(props) =>
+    props.size === 'lg' ? `1.5rem 4rem` : `0.875rem 1.5rem`};
+  background: ${(props) =>
+    props.disabled
+      ? color(props.background)
```
Comment (hedvigoscar) [inline src/new-components/buttons.tsx]: :eyes: Quite nice actually

File: src/pages/OfferNew/Compare/index.tsx [hunk 3]
```diff
@@ -23,7 +25,13 @@ export const Compare = () => {
         <Column>
           <HeadingWrapper>
             <PreHeading>Villkor</PreHeading>
-            <HeadingBlack>Jämför gärna!</HeadingBlack>
+            <HeadingBlack>
+              Allt du förväntar dig av en hemförsäkring, plus drulle
```
Comment (hedvigoscar) [inline src/pages/OfferNew/Compare/index.tsx]: We actually have like all the text keys available, I think we're going to need to get them in use before too long 😬

File: src/pages/OfferNew/Introduction/Sidebar/mock.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { InsuranceType } from 'utils/insuranceDomainUtils'
+
+export interface InsuranceCompany {
+  name: string
+  switchable: boolean
+}
+
+export const otherInsuranceCompanies: InsuranceCompany[] = [
+  { name: 'Folksam', switchable: true },
```
Comment (hedvigoscar) [inline src/pages/OfferNew/Introduction/Sidebar/mock.ts]: We might be able to use the `PreviousInsurer`-type here if we're really, really lucky. I'll investigate and get back to you

File: src/utils/hooks/useDocumentScroll.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react'
+
+export const useDocumentScroll = (
```
[the remark this answers is not in the document]
Comment (gustafguner) [reply src/utils/hooks/useDocumentScroll.tsx]: React Hooks <3

File: src/pages/OfferNew/Introduction/Sidebar/mock.ts (hunk 4 above)
Comment (gustafguner) [reply src/pages/OfferNew/Introduction/Sidebar/mock.ts]: Nice! Yes, we'll leave it like this for now and fix this when we know how the previous insurer will be handled!

File: src/pages/OfferNew/Compare/index.tsx (hunk 3 above)
Comment (gustafguner) [reply src/pages/OfferNew/Compare/index.tsx]: Very nice, let's go though all hard-coded copy when this is merged 😄

File: src/new-components/buttons.tsx (hunk 2 above)
Comment (gustafguner) [reply src/new-components/buttons.tsx]: Ikr! 🎨

File: src/new-components/Tooltip.tsx (hunk 1 above)
Comment (gustafguner) [reply src/new-components/Tooltip.tsx]: Yes, much more neat 👌

**Pull request merged**